cli/utils.js

```diff
@@ -253,11 +253,45 @@ const loadBlueprintsFromYoRc = () => {
     return loadBlueprintsFromConfiguration(yoRc['generator-jhipster']);
 };
 
+/**
+ * Creates a 'blueprintName: blueprintVersion' object.
+ */
+const loadAllBlueprintsWithVersion = () => {
+    const blueprintsWithVersion = loadBlueprints().reduce((acc, blueprint) => {
+        acc[blueprint] = undefined;
+        return acc;
+    }, {});
+
+    loadBlueprintsFromYoRc().reduce((acc, blueprint) => {
+        acc[blueprint.name] = blueprint.version;
+        return acc;
+    }, blueprintsWithVersion);
+    return blueprintsWithVersion;
+};
```

Comment:
this needs some testing

Reply:
There isn't test for loadBlueprints, loadYoRc and loadBlueprintsFromConfiguration.
Mocking them is not possible due to using them as local variables.

loadBlueprints is subject to change due to commander 5 . https://github.com/jhipster/generator-jhipster/issues/11670

For hipster 7 I want to change the app sub generator to another generator and move most of the cli logic to app generator.
IMO it's too complex right now.
So add testing to them it's not easy and IMO is not worth now.

Reply:
I've written tests for that, please have a look :)